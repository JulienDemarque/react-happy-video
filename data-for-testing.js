export const ProgressBarProps = {
	colorgiven: '#3b3346',
	progress: 0.2
};

export const SpeedControlProps = {
	speed: 1
};

export const TimeDisplayProps = {
	time: 20,
	duration: 300
};

export const VideoPlayerProps = {
	width: '100%',
	source: 'www.mymockurl.com/mymockvideo.mp4',
	color: '#3b3346'
};

export const VolumeControlProps = {
	volume: 0.5
};
