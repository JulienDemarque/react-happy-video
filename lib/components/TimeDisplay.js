import React from 'react';
import { formatSeconds } from '../utils/format-seconds';
import { Container, SubContainer } from '../styles/components/time-display';

class TimeDisplay extends React.Component {
	constructor(props) {
		super(props);
	}
	render() {
		return (
			<Container>
				<SubContainer>
					<p>
						{formatSeconds(this.props.time)} /{' '}
						{formatSeconds(this.props.duration)}
					</p>
				</SubContainer>
			</Container>
		);
	}
}

export default TimeDisplay;
