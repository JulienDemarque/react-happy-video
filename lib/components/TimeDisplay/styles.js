import styled from 'styled-components';

export const SubContainer = styled.div`
	display: flex;
	align-items: center;
	padding: 0 20px;
`;
