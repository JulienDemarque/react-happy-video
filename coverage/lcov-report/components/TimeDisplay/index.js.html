<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/TimeDisplay/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/TimeDisplay</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { formatSeconds } from '../../utils/format-seconds';
import { SubContainer } from './styles';
import PropTypes from 'prop-types';
&nbsp;
class TimeDisplay extends React.PureComponent {
	constructor(props) {
		super(props);
	}
	render() {
		const { time, duration } = this.props;
		return (
			&lt;SubContainer&gt;
				&lt;p&gt;
					{formatSeconds(time)} / {formatSeconds(duration)}
				&lt;/p&gt;
			&lt;/SubContainer&gt;
		);
	}
}
&nbsp;
TimeDisplay.propTypes = {
	time: PropTypes.number.isRequired,
	duration: PropTypes.number.isRequired
};
&nbsp;
export default TimeDisplay;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 05 2018 20:47:31 GMT+0200 (IST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
