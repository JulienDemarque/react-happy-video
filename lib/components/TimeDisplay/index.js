import React from 'react';
import { formatSeconds } from '../../utils/format-seconds';
import { SubContainer } from './styles';
import PropTypes from 'prop-types';

class TimeDisplay extends React.PureComponent {
	constructor(props) {
		super(props);
	}
	render() {
		const { time, duration } = this.props;
		return (
			<SubContainer>
				<p>
					{formatSeconds(time)} / {formatSeconds(duration)}
				</p>
			</SubContainer>
		);
	}
}

TimeDisplay.propTypes = {
	time: PropTypes.number.isRequired,
	duration: PropTypes.number.isRequired
};

export default TimeDisplay;
