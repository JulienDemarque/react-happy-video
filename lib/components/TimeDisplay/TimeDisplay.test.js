import React from 'react';
import { shallow } from 'enzyme';
import TimeDisplay from './index';
import { TimeDisplayProps } from '../../../data-for-testing';

const { time, duration } = TimeDisplayProps;

describe('TimeDisplay Component', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<TimeDisplay time={time} duration={duration} />);
	});

	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
