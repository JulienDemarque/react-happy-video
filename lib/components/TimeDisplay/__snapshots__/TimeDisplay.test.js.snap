// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeDisplay Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimeDisplay
    duration={300}
    time={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <p>
        0:20
         / 
        5:00
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "0:20",
          " / ",
          "5:00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "0:20",
        " / ",
        "5:00",
      ],
      "type": "p",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
	display: flex;
	align-items: center;
	padding: 0 20px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          0:20
           / 
          5:00
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "0:20",
            " / ",
            "5:00",
          ],
        },
        "ref": null,
        "rendered": Array [
          "0:20",
          " / ",
          "5:00",
        ],
        "type": "p",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
	display: flex;
	align-items: center;
	padding: 0 20px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
